### ISSUE TITLE
Cisco ASA Legacy connector still receiving logs after migration to AMA connector

### ISSUE THREAD

**Issue by thom2804** [Azure/Azure-Sentinel#10937]
Hello everyone, so recently I have migrated my machines from an environment with Cisco ASA from the Legacy Connector to the new AMA connector.
I uninstalled the legacy connector completely from my machine, and the Workspace settings for the Sentinel environment also show there are no more legacy agents connected or collections rules active for these.
However, when I go to the data connector section, both the legacy and AMA connector still show up as connected with both receiving the same amount of data.
I would like to disconnect and remove the legacy connector, however I can't figure out how to do this, can anyone help with this?

Note: I am receiving all logs properly from the new AMA connector and there is no other data being collected as far as I can see, it just seems like the legacy and the AMA connector reference the same data table causing this issue.

**Comment by v-sudkharat** [comment]
Hi @thom2804, Thanks for flagging this issue, we will investigate this issue and get back to you with some updates. Thanks!

**Comment by thom2804** [comment]
Are there any updates about this?
To this day I am still receiving logging in the legacy dataconnector holding me back from being able to delete the dataconnector.
Maybe there needs to be something changed in the parser to exclude the AMA logs in the legacy connector?

**Comment by v-rusraut** [comment]
Hi @thom2804,

We have updated template to fix this issue , could you please custom deploy the Solution MainTemplate.

Sharing the MainTemplate and Custom deployment steps-
Custom Deployment - [CustomDeploymentSteps.docx](https://github.com/user-attachments/files/16782483/CustomDeploymentSteps.docx)
mainTemplate - [mainTemplate.json](https://github.com/user-attachments/files/16844264/mainTemplate.json)

After custom deploy the mainTemplate ,you will see legacy data connector showing as disconnected.

Then follow below steps to delete this data connector.

- Search cisco ASA Legacy dataconnector in Data Connector section.

![image](https://github.com/user-attachments/assets/5eef9c96-8c4d-461a-bf41-846ff5f84fff)

-  Click on delete button to delete data connector.

![image](https://github.com/user-attachments/assets/75b05c2a-c2e5-4cb8-b1e3-ab259d78c6b8)

- Please confirm delete , after that legacy data connector will be deleted.

![image](https://github.com/user-attachments/assets/a9fbd086-6014-480e-87d2-4d7c814b5324)

Please let us know that your issue resolved or not.

Thanks

**Comment by thom2804** [comment]
Hi @v-rusraut thanks for the quick response, I performed the steps and after doing so the solution showed that the legacy connector is deleted and that I can delete it successfully.
The AMA connector was still active along with the DCR rule I created for this.
Would this same method by removing and reinstalling the solution also work for other solutions which suffer from the same issue? (for example Fortinet)
